tasks/system_probe.py

```diff
@@ -927,7 +927,14 @@ def kitchen_prepare(ctx, kernel_release=None, ci=False, packages=""):
         if pkg.endswith("java"):
             shutil.copy(os.path.join(pkg, "agent-usm.jar"), os.path.join(target_path, "agent-usm.jar"))
 
-        for gobin in ["gotls_client", "grpc_external_server", "external_unix_proxy_server", "fmapper", "prefetch_file"]:
+        for gobin in [
+            "external_unix_proxy_server",
+            "fmapper",
+            "gotls_client",
+            "gotls_server",
```

Comment:
Wondering if using the `external_unix_proxy_server` would have make the new `gotls_server` binary, redundant
What do you think?

Reply:
Well, the `gotls_server` was made to see if the issue we have with the Python server reproduced with a Go server, and as such, I wanted the overall flow of the test to be as close as possible: the client is the test binary, the server is an external one. I did consider at some point using the proxy server, but that shifted this flow a bit. Unless it is possible to have the HTTP server to listen to the Unix socket of the proxy?

The proxy server shifts it to: the client and server are in the test binary, but the client relies on the proxy to send its requests to the server and handle encryption.

Now, the goal of the test is to classify a HTTPS connection. And in that regard, I think both options work fine. I can change the test to use the proxy instead, I don't mind it. But then I think we should also discuss if `gotls_client` isn't redundant.